src/govuk/components/accordion/accordion.test.js

```diff
@@ -230,32 +231,57 @@ describe('/components/accordion', () => {
         })
       })
 
-      it('should change the (combined) Show[ this section] text to Hide[ this section ] when sections are opened', async () => {
+      it('should change the Show text to Hide when sections are opened', async () => {
         await goToComponent(page, 'accordion')
         await page.click('.govuk-accordion .govuk-accordion__section:nth-of-type(2) .govuk-accordion__section-header')
         await page.click('.govuk-accordion .govuk-accordion__section:nth-of-type(3) .govuk-accordion__section-header')
 
         const ShowOrHideButtonText = await page.evaluate(() => document.body.querySelector('.govuk-accordion__section-toggle-text').textContent)
 
-        expect(ShowOrHideButtonText).toEqual('Hide this section')
+        expect(ShowOrHideButtonText).toEqual('Hide')
       })
 
-      it('should have a data-nosnippet attribute on the "Show / hide this section" container to hide it from search result snippets', async () => {
+      it('should have a data-nosnippet attribute on the "Show / hide" container to hide it from search result snippets', async () => {
         await goToComponent(page, 'accordion')
 
         const dataNoSnippetAttribute = await page.evaluate(() => document.body.querySelector('.govuk-accordion__section-toggle').getAttribute('data-nosnippet'))
 
         expect(dataNoSnippetAttribute).toEqual('')
       })
 
-      describe('hidden text', () => {
-        it('should contain hidden text " this section" for assistive technology', async () => {
+      describe('accessible name', () => {
+        it('Should set the accessible name of the show/hide section buttons', async () => {
           await goToComponent(page, 'accordion')
 
-          const numberOfExampleSections = 2
-          const hiddenText = await page.evaluate(() => document.body.querySelectorAll('.govuk-accordion .govuk-accordion__section .govuk-accordion__section-toggle-text .govuk-visually-hidden').length)
+          const node = await page.$(
+            'aria/Section A , Show this section[role="button"]'
+          )
+          expect(node).not.toBeNull()
+
+          await node.click()
+
+          const updatedNode = await page.$(
```

Comment:
A lot of the other tests are using longer `document.body.querySelector` from before we knew about the `$` and `$$` methods so feel free to simplify them too.

Don't forget we have a new [`getAttribute()` helper](https://github.com/alphagov/govuk-frontend/blob/main/lib/puppeteer-helpers.js#L121) for **aria-label** as well 🎉

Reply:
It's important we rely on the `aria` selector for that test as the aim is to check what accessible name gets provided to assistive technology, and not take the responsibility of computing it ourselves depending on what we provide as [there are quite a few possibilities, like the inside of axe-core show](https://github.com/dequelabs/axe-core/blob/006a681395422bbd0603bab346dbdc6b38087d83/lib/commons/text/accessible-text-virtual.js#L29-L37) (no luck on that side finding an easy computation of the accessible name, though).

Unfortunately, in its current form, Puppeteer does not provide a `Element.getAccesibleTreeNode()` (invented name)
 that would let us cache the first found node to check that its corresponding bit in the accessibility tree has changed how we desire. ([fingers crossed that'll land in the future](https://github.com/puppeteer/puppeteer/issues/3641)). So we do need to re-query for that second ElementHandle[^1]
 
The risk of confusion is pretty limited, I think, given we're checking a distinct label "Section A" in the accessible name alongside the role `[role="button"]`. Given the example it runs onto and what we expect of our component, we know it's a unique thing. I could check for the absence of the other button but I don't think it adds much.

[^1]: Only other access to the accessibility tree information is through `page.accessibility.snapshot()`, but same issue we'd have to snapshot twice, and on top of that, do the same kind of walk Puppeteer must be doing when running its selectors.

Reply:
@romaricpascal I tried the same, to see if a computed name was available 😭 

But yeah still test for the accessible name, I like it 👍 

The concern was solely that the new test passes with duplicate buttons on the page:

**Button 1** `aria/Section A , Hide this section[role="button"]`
**Button 2** `aria/Section A , Show this section[role="button"]`

We're not testing that it's the same button with toggled text

Reply:
The examples we run on ensure the uniqueness, but I had a further dig to see if we could be safe in that respect anyway. 

Turns out you can get the accessible name from an `ElementHandle`, it's just not on the `ElementHandle` itself 🎉  `page.accessibility.snapshot()` accepts a `root` option to get a snapshot starting at a given node in the accessibility tree. Not all nodes will have an entry in the accessibility tree, so it may return null.

I've [updated the tests](https://github.com/alphagov/govuk-frontend/pull/2932/files#diff-074bb9b59bea515d21a2ba82e0e9f1cb6f0959302b6b9ea41c1c76cc80b31162R258-R284) which work more like you'd expect them to (check accessibility name, take action, check accessibility name) and [encapsulated the access to the tree and accessibility name in helpers](https://github.com/alphagov/govuk-frontend/pull/2932/files#diff-e8850c0e27106542fde2e15df546e04781301f4c63d9f887f56ac80c9e641a3bR125-R153) as it's something we'll likely want to use in other tests.

Reply:
Thanks @romaricpascal, absolutely perfect

Means we're checking the same element so I'm happy